docs/en/latest/practices/the-hard-way.md

```diff
@@ -0,0 +1,826 @@
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - networking.k8s.io
+    resources:
+      - ingresses
+      - networkpolicies
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - metrics.k8s.io
+    resources:
+      - pods
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - apisix.apache.org
+    resources:
+      - apisixroutes
+      - apisixupstreams
+      - apisixservices
+      - apisixtlses
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - coordination.k8s.io
+    resources:
+      - leases
+    verbs:
+      - '*'
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: apisix-clusterrolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: apisix-clusterrole
+subjects:
+  - kind: ServiceAccount
+    name: apisix-ingress-controller
+    namespace: apisix
+```
+
+Then, we need to create ApisixRoute CRD:
+
+```yaml
+
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: apisixroutes.apisix.apache.org
+spec:
+  group: apisix.apache.org
+  versions:
+    - name: v1
+      served: true
+      storage: false
+    - name: v2alpha1
+      served: true
+      storage: true
+  scope: Namespaced
+  names:
+    plural: apisixroutes
+    singular: apisixroute
+    kind: ApisixRoute
+    shortNames:
+      - ar
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: apisixtlses.apisix.apache.org
+spec:
+  group: apisix.apache.org
+  versions:
+    - name: v1
+      served: true
+      storage: true
+  scope: Namespaced
+  names:
+    plural: apisixtlses
+    singular: apisixtls
+    kind: ApisixTls
+    shortNames:
+      - atls
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: apisixupstreams.apisix.apache.org
+spec:
+  group: apisix.apache.org
+  versions:
+    - name: v1
+      served: true
+      storage: true
+  scope: Namespaced
+  names:
+    plural: apisixupstreams
+    singular: apisixupstream
+    kind: ApisixUpstream
+    shortNames:
+      - au
+```
+
+To make the ingress controller works properly with APISIX, we need to create a config file containing the APISIX admin API URL and API key as below:
+
+```yaml
+apiVersion: v1
+data:
+  config.yaml: |
+    # log options
+    log_level: "debug"
+    log_output: "stderr"
+    http_listen: ":8080"
+    enable_profiling: true
+    kubernetes:
+      kubeconfig: ""
+      resync_interval: "30s"
+      app_namespaces:
+      - "*"
+      ingress_class: "apisix"
+      ingress_version: "networking/v1"
+      apisix_route_version: "apisix.apache.org/v1"
+    apisix:
+      base_url: "http://apisix-admin.apisix:9180/apisix/admin"
+      admin_key: "edd1c9f034335f136f87ad84b625c8f1"
+kind: ConfigMap
+metadata:
+  name: apisix-configmap
+  labels:
+    app.kubernetes.io/name: ingress-controller
+```
```

Comment:
Add a note that if you want to learn all the configuration items, see [conf/config-default.yaml](./conf/config-default.yaml) for details.